tests/Functional/Action/Lti/UpdateLti1p3OutcomeActionTest.php

```diff
@@ -0,0 +1,241 @@
+<?php
+
+/**
+ *  This program is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU General Public License
+ *  as published by the Free Software Foundation; under version 2
+ *  of the License (non-upgradable).
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ *  Copyright (c) 2019 (original work) Open Assessment Technologies S.A.
+ */
+
+declare(strict_types=1);
+
+namespace OAT\SimpleRoster\Tests\Functional\Action\Lti;
+
+use OAT\Bundle\Lti1p3Bundle\Tests\Traits\SecurityTestingTrait;
+use OAT\Library\Lti1p3Core\Registration\RegistrationInterface;
+use OAT\Library\Lti1p3Core\Registration\RegistrationRepositoryInterface;
+use OAT\SimpleRoster\Entity\Assignment;
+use OAT\SimpleRoster\Repository\AssignmentRepository;
+use OAT\SimpleRoster\Tests\Traits\DatabaseTestingTrait;
+use Ramsey\Uuid\Uuid;
+use Ramsey\Uuid\UuidFactoryInterface;
+use Ramsey\Uuid\UuidInterface;
+use Symfony\Bundle\FrameworkBundle\KernelBrowser;
+use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
+use Symfony\Component\HttpFoundation\Response;
+
+class UpdateLti1p3OutcomeActionTest extends WebTestCase
+{
+    use DatabaseTestingTrait;
+    use SecurityTestingTrait;
+
+    /** @var KernelBrowser */
+    private $kernelBrowser;
+
+    /** @var RegistrationInterface */
+    private $registration;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->kernelBrowser = self::createClient();
+        $this->setUpDatabase();
+        $this->loadFixtureByFilename('userWithReadyAssignment.yml');
+
+        /** @phpstan-ignore-next-line */
+        $this->registration = static::$container
+            ->get(RegistrationRepositoryInterface::class)
+            ->find('testRegistration');
+    }
+
+    public function testItReturns401IfNotAuthenticated(): void
+    {
+        $this->kernelBrowser->request('POST', '/api/v1/lti1p3/outcome');
+
+        self::assertEquals(Response::HTTP_UNAUTHORIZED, $this->kernelBrowser->getResponse()->getStatusCode());
+    }
+
+    public function testItReturns200IfTheAuthenticationWorksAndAssignmentExists(): void
+    {
+        $credentials = $this->createTestClientAccessToken(
+            $this->registration,
+            ['https://purl.imsglobal.org/spec/lti-bo/scope/basicoutcome']
+        );
+
+        $uidGenerator = $this->createMock(UuidFactoryInterface::class);
+        self::$container->set('test.uid_generator', $uidGenerator);
+
+        $uidGenerator
```

Comment:
I believe that you do not need this identation

Reply:
Kept as is for now, we can discuss it with the team